python.d/postgres.chart.py

```diff
@@ -21,222 +21,358 @@
-                'vacuum_freeze',
-                'brin_summarize'],
-    STANDBY_DELTA=['sent_delta',
-                   'write_delta',
-                   'flush_delta',
-                   'replay_delta'
-                   ],
-    REPSLOT_FILES=['replslot_wal_keep',
-                   'replslot_files']
-
-)
+METRICS = {
+    'DATABASE': [
+        'connections',
+        'xact_commit',
+        'xact_rollback',
+        'blks_read',
+        'blks_hit',
+        'tup_returned',
+        'tup_fetched',
+        'tup_inserted',
+        'tup_updated',
+        'tup_deleted',
+        'conflicts',
+        'temp_files',
+        'temp_bytes',
+        'size'
+    ],
+    'BACKENDS': [
+        'backends_active',
+        'backends_idle'
+    ],
+    'INDEX_STATS': [
+        'index_count',
+        'index_size'
+    ],
+    'TABLE_STATS': [
+        'table_size',
+        'table_count'
+    ],
+    'WAL': [
+        'written_wal',
+        'recycled_wal',
+        'total_wal'
+    ],
+    'WAL_WRITES': [
+        'wal_writes'
+    ],
+    'ARCHIVE': [
+        'ready_count',
+        'done_count',
+        'file_count'
+    ],
+    'BGWRITER': [
+        'checkpoint_scheduled',
+        'checkpoint_requested',
+        'buffers_checkpoint',
+        'buffers_clean',
+        'maxwritten_clean',
+        'buffers_backend',
+        'buffers_alloc',
+        'buffers_backend_fsync'
+    ],
+    'LOCKS': [
+        'ExclusiveLock',
+        'RowShareLock',
+        'SIReadLock',
+        'ShareUpdateExclusiveLock',
+        'AccessExclusiveLock',
+        'AccessShareLock',
+        'ShareRowExclusiveLock',
+        'ShareLock',
+        'RowExclusiveLock'
+    ],
+    'AUTOVACUUM': [
+        'analyze',
+        'vacuum_analyze',
+        'vacuum',
+        'vacuum_freeze',
+        'brin_summarize'
+    ],
+    'STANDBY_DELTA': [
+        'sent_delta',
+        'write_delta',
+        'flush_delta',
+        'replay_delta'
+    ],
+    'REPSLOT_FILES': [
+        'replslot_wal_keep',
+        'replslot_files'
+    ]
+}
 
-QUERIES = dict(
-    WAL="""
+QUERIES = {
+    'WAL': """
 SELECT
-  count(*) as total_wal,
-  count(*) FILTER (WHERE type = 'recycled') AS recycled_wal,
-  count(*) FILTER (WHERE type = 'written') AS written_wal
-FROM
-  (SELECT wal.name,
-     pg_{0}file_name(CASE pg_is_in_recovery() WHEN true THEN NULL ELSE pg_current_{0}_{1}() END ),
-     CASE WHEN wal.name > pg_{0}file_name(CASE pg_is_in_recovery() WHEN true THEN NULL ELSE pg_current_{0}_{1}() END ) THEN 'recycled'
-       ELSE 'written'
-     END AS type
-  FROM pg_catalog.pg_ls_dir('pg_{0}') AS wal(name)
-  WHERE name ~ '^[0-9A-F]{{24}}$'
-  ORDER BY (pg_stat_file('pg_{0}/'||name)).modification, wal.name DESC) sub;
+    count(*) AS total_wal,
+    count(*) FILTER (
+    WHERE
+        TYPE = 'recycled') AS recycled_wal, count(*) FILTER (
+    WHERE
+        TYPE = 'written') AS written_wal
+FROM (
+    SELECT
+        wal.name, pg_ 0 file_name (
```

Comment:
Last commit has broken queries. Here `pg_{0}file_name` has been changed to `pg_ 0 file_name`